### PR TITLE
[better-sqlite3] Add type declarations

### PR DESCRIPTION
**Pull request by Morfent**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

Round 2. @joshuawise, can you check if there's anything missing or mistyped?

### CODE REVIEW

**Comment by Morfent** [comment]
The commit was missing an author earlier. Fixed

File: types/better-sqlite3/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+interface DatabaseOptions {
+	memory?: boolean;
+	readonly?: boolean;
+}
+
+interface RegistrationOptions {
+	name?: string;
+	varargs?: boolean;
+	deterministic?: boolean;
```
Comment (JoshuaWise) [inline types/better-sqlite3/index.d.ts]: this is missing `safeIntegers?: boolean`

File: types/better-sqlite3/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+// Definitions by: Ben Davies <https://github.com/Morfent/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import * as Integer from 'integer';
+
+interface RunResult {
+	changes: number;
+	lastInsertRowID: number;
```
Comment (JoshuaWise) [inline types/better-sqlite3/index.d.ts]: this is misspelled, it should be `lastInsertROWID`, and it can be a `number` or a `Database.Integer`

**Comment by Morfent** [comment]
Fixed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/19145

**Pull request merged**